### DISCUSSION TITLE
Question about graphing function in IDE

### DISCUSSION THREAD

**Discussion by meistef** [MichaelClerx/myokit#1027] [Q&A] [answered]
Hello,

What protocol is being run if you use 'command + G' on a line in the myokit IDE and how do I re-create this? I quickly use that option to evaluate currents (for example IK1 in ORd) after I fixed the Nernst potential. It plots IK1 automatically over a range of -100 mV to +100 mV in a continuous fashion.

**Comment by MichaelClerx** [comment, marked as answer]
Hi Stefan!

The graphing function actually doesn't do any simulation, but uses the [pyfunc()](https://myokit.readthedocs.io/en/stable/api_core/expressions.html#myokit.Variable.pyfunc) method to turn a myokit equation into a callable python function.
You can do the same using code like this:

```
[[script]]
import myokit

import matplotlib.pyplot as plt
import numpy as np

# Create model
m = myokit.parse_model(
'''
[[model]]
membrane.V = 0

[membrane]
time = 0 [ms]
    in [ms]
    bind time
dot(V) = 0 [mV/ms]
    in [mV]

[ik1]
use membrane.V
E_K = -85 [mV]
    in [mV]
inf = 1 / (1 + exp(0.1 [1/mV] * (V + 75.44 [mV])))
g_K1 = 0.065 [mS/uF]
    in [mS/uF]
i_K1 = g_K1 * inf * (V - 8 [mV] - E_K)
    in [A/F]
''')

# Get IK1 variable from model
ik1 = m.get('ik1.i_K1')

# Get numpy-ready callable function, with state variable V as argument
f = ik1.pyfunc(use_numpy=True)

# Plot for range of V
v = np.linspace(-120, 0, 121)


fig = plt.figure()
ax = fig.add_subplot()
ax.axhline(0, color='#ccc')
ax.set_xlabel('Vm (mV)')
ax.set_ylabel('IK1 (A/F)')
ax.plot(v, f(v))
plt.show()
```

![stefan-ik1](https://github.com/myokit/myokit/assets/517644/9a6f5e4f-1a54-47f3-87fe-b577507b5710)


Drawing a nice I-V like this will only work for currents (or any other variables) that rely on no other state-variables than V (so no HH gating or Markov states, or dynamic [Ca] dependency etc). For those you will need simulation, like in the 2nd Beeler-Reuter example here: http://myokit.org/examples/#br-1977
